test/tests/functional/pbs_reliable_job_startup.py

```diff
@@ -5374,53 +5374,56 @@ def test_t20(self):
-                exec_vnode = self.jobA_exec_vnode2
-                exec_vnode_esc = self.jobA_exec_vnode2_esc
-                vnode_list = [self.nAv1, self.nBv0, self.nC,
-                              self.nD, self.nEv0]
+                d = {'iexec_host_esc' : self.jobA_iexec_host2_esc,
+                'iexec_vnode' : self.jobA_iexec_vnode2,
+                'iexec_vnode_esc' : self.jobA_iexec_vnode2_esc,
+                'exec_host' : self.jobA_exec_host2,
+                'exec_host_esc' : self.jobA_exec_host2_esc,
+                'exec_vnode' : self.jobA_exec_vnode2,
+                'exec_vnode_esc' : self.jobA_exec_vnode2_esc,
+                'vnode_list' : [self.nAv1, self.nBv0, self.nC,
+                              self.nD, self.nEv0]}
             elif idx == 3:
-                iexec_host_esc = self.jobA_iexec_host3_esc
-                iexec_vnode = self.jobA_iexec_vnode3
-                iexec_vnode_esc = self.jobA_iexec_vnode3_esc
-                exec_host = self.jobA_exec_host3
-                exec_host_esc = self.jobA_exec_host3_esc
-                exec_vnode = self.jobA_exec_vnode3
-                exec_vnode_esc = self.jobA_exec_vnode3_esc
-                vnode_list = [self.nAv2, self.nBv1, self.nC,
-                              self.nD, self.nE]
-
+                d = {'iexec_host_esc' : self.jobA_iexec_host3_esc,
+                'iexec_vnode' : self.jobA_iexec_vnode3,
+                'iexec_vnode_esc' : self.jobA_iexec_vnode3_esc,
+                'exec_host' : self.jobA_exec_host3,
+                'exec_host_esc' : self.jobA_exec_host3_esc,
+                'exec_vnode' : self.jobA_exec_vnode3,
+                'exec_vnode_esc' : self.jobA_exec_vnode3_esc,
+                'vnode_list' : [self.nAv2, self.nBv1, self.nC,
+                              self.nD, self.nE]}
+            idx_values[idx] = d
             self.server.expect(JOB, {'job_state': 'R',
                                      'substate': 41,
                                      'tolerate_node_failures': 'job_start',
                                      'Resource_List.mem': '3gb',
                                      'Resource_List.ncpus': 3,
                                      'Resource_List.nodect': 3,
-                                     'exec_host': exec_host,
-                                     'exec_vnode': exec_vnode,
+                                     'exec_host': d['exec_host'],
+                                     'exec_vnode': d['exec_vnode'],
                                      'Resource_List.select': self.jobA_select,
                                      'Resource_List.site': self.jobA_oselect,
                                      'Resource_List.place': self.jobA_place,
                                      'schedselect': self.jobA_schedselect},
                                id=sjid, attrop=PTL_AND)
 
+        for idx in range(1, 4):
+            sjid = create_subjob_id(jid, idx)
```

Comment:
Rather than calling create_subjob_id() again here, you can just cache the sjid values in an array, like adding something after line 5411:
 sjid_values[idx] = sjid, 
then can just put in here:
 sjid = sjid_values[idx].